include/ginkgo/core/log/batch_logger.hpp

```diff
@@ -33,12 +34,11 @@ namespace detail {
 template <typename ValueType>
 struct log_data final {
     using real_type = remove_complex<ValueType>;
+    using idx_type = int;
 
     log_data(std::shared_ptr<const Executor> exec, size_type num_batch_items)
```

Comment:
all of this implementation can be moved to core. Then we don't need the CUB file in our public header.

Reply:
Unfortunately, I dont think this can be moved to core.